### PR TITLE
CORDA-716 Retire MockNetwork.Factory

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: core/src/test/kotlin/net/corda/core/flows/AttachmentTests.kt [hunk 1]
```diff
@@ -109,13 +108,11 @@ class AttachmentTests {
     }
 
     @Test
-    fun `malicious response`() {
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/flows/AttachmentTests.kt]: this was causing a kotlin compile error and i'm afraid to change it back

Comment (mnesbit) [reply core/src/test/kotlin/net/corda/core/flows/AttachmentTests.kt]: That's fine.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 2]
```diff
@@ -65,13 +65,13 @@ data class MockNetworkParameters(
...
         val servicePeerAllocationStrategy: InMemoryMessagingNetwork.ServicePeerAllocationStrategy = InMemoryMessagingNetwork.ServicePeerAllocationStrategy.Random(),
-        val defaultFactory: MockNetwork.Factory<*> = MockNetwork.DefaultFactory,
+        val defaultFactory: (MockNodeArgs) -> MockNetwork.MockNode = MockNetwork::MockNode,
         val initialiseSerialization: Boolean = true,
         val cordappPackages: List<String> = emptyList()) {
     fun setNetworkSendManuallyPumped(networkSendManuallyPumped: Boolean) = copy(networkSendManuallyPumped = networkSendManuallyPumped)
     fun setThreadPerNode(threadPerNode: Boolean) = copy(threadPerNode = threadPerNode)
     fun setServicePeerAllocationStrategy(servicePeerAllocationStrategy: InMemoryMessagingNetwork.ServicePeerAllocationStrategy) = copy(servicePeerAllocationStrategy = servicePeerAllocationStrategy)
-    fun setDefaultFactory(defaultFactory: MockNetwork.Factory<*>) = copy(defaultFactory = defaultFactory)
+    fun setDefaultFactory(defaultFactory: (MockNodeArgs) -> MockNetwork.MockNode) = copy(defaultFactory = defaultFactory)
```
Comment (mnesbit) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Is there a risk with this that the Java usage is nasty? We eliminated Kotlin lambdas from our API, do we need to take care with the test api?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 3]
```diff
@@ -121,7 +121,7 @@ class MockNetwork(defaultParameters: MockNetworkParameters = MockNetworkParamete
                   private val networkSendManuallyPumped: Boolean = defaultParameters.networkSendManuallyPumped,
                   private val threadPerNode: Boolean = defaultParameters.threadPerNode,
                   servicePeerAllocationStrategy: InMemoryMessagingNetwork.ServicePeerAllocationStrategy = defaultParameters.servicePeerAllocationStrategy,
-                  private val defaultFactory: Factory<*> = defaultParameters.defaultFactory,
+                  private val defaultFactory: (MockNodeArgs) -> MockNode = defaultParameters.defaultFactory,
```
Comment (mnesbit) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Kotlin lambda?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt (hunk 2 above)
Comment (andr3ej) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNode.kt]: i've now added MockNodeFactoryInJavaTest to enforce a nice java experience

**Pull request merged**